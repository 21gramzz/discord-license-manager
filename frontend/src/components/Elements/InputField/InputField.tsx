import React from 'react';
import styled from 'styled-components';

export interface InputProps {
  className?: string;
  id?: string;
  name?: string;
  placeholder?: string;
  type?: string;
  color?: string;
  backgroundColor?: string;
  width?: string | number;
  height?: string | number;
  margin?: string;
  padding?: string;
  value?: string | number;
  step?: string;
  min?: string | number;
  max?: string | number;
  defaultValue?: string | number;
  fontSize?: string;
  fontWeight?: string;
  inputRef?: any;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onBlur?: React.FocusEventHandler<HTMLInputElement>;
}

export const InputField: React.FC<InputProps> = ({ inputRef, ...props }) => {
  return <StyledInputField ref={inputRef} {...props} />;
};

const StyledInputField = styled.input<
  Pick<
    InputProps,
    | 'margin'
    | 'padding'
    | 'color'
    | 'backgroundColor'
    | 'width'
    | 'height'
    | 'fontSize'
    | 'fontWeight'
  >
>`
  color: ${({ color, theme }) => color || theme.text};
  width: ${({ width }) => width || 'auto'};
  height: ${({ height }) => height || '3.5rem'};
  margin: ${({ margin }) => margin || '0'};
  background-color: ${({ backgroundColor }) =>
    backgroundColor || 'transparent'};
  padding: ${({ padding }) => padding || '0.5rem 1.5rem'};
  outline: 0;
  border: ${({ theme }) => 'solid 0.5px ' + theme.border};
  font-weight: ${({ fontWeight }) => fontWeight || '400'};
  font-size: ${({ fontSize }) => fontSize || '1.3rem'};
  letter-spacing: 0.025em;
  border-radius: 0.5rem;
  ::placeholder {
    color: ${({ theme }) => theme.border};
  }
  &.is-invalid {
    border: solid 1px ${({ theme }) => theme.danger};
  }
`;
