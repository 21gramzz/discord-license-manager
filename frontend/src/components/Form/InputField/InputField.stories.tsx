import { ComponentMeta, ComponentStoryObj } from '@storybook/react';
import { InputField } from '.';

export default { component: InputField } as ComponentMeta<typeof InputField>;

export const Default: ComponentStoryObj<typeof InputField> = {
  args: {
    placeholder: 'Type here',
  },
};

export const Error: ComponentStoryObj<typeof InputField> = {
  args: {
    placeholder: 'Type here',
    className: 'is-invalid',
  },
};
