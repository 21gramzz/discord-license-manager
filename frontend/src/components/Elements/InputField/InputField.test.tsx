import initStoryshots from '@storybook/addon-storyshots';
import * as stories from './InputField.stories';
import { composeStories } from '@storybook/testing-react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

initStoryshots();

describe('InputField', () => {
  const { Default } = composeStories(stories);

  test('should render default input field', async () => {
    render(<Default placeholder="Type here" />);
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });

  test('should call onChange event', async () => {
    const onChange = jest.fn();
    render(<Default onChange={onChange} />);
    fireEvent.change(screen.getByRole('textbox'), {
      target: { value: 'test' },
    });
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
