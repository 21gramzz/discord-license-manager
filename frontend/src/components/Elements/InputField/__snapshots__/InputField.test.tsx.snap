// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots  InputField Default 1`] = `
<input
  className="sc-bczRLJ fAQBvE"
  placeholder="Type here"
/>
`;

exports[`Storyshots  InputField Error 1`] = `
<input
  className="sc-bczRLJ fAQBvE is-invalid"
  placeholder="Type here"
/>
`;

exports[`Storyshots Elements/Button Default 1`] = `
<button
  className="sc-gsnTZi cFnfUp"
>
  Default Button
</button>
`;

exports[`Storyshots Elements/Button Primary 1`] = `
<button
  className="sc-gsnTZi sc-dkzDqf cFnfUp cGQMqD"
>
  Primary Button
</button>
`;
